### DISCUSSION TITLE
Drawing is not working, I have use given example

### DISCUSSION THREAD

**Discussion by Anubhavshakya** [BenJeau/react-native-draw#45] [Q&A] [answered]
Please provide correct example to use it. In my case UI is rendering but it does not draw anything.

**Comment by BenJeau** [comment]
I am assuming that you have not properly setup https://docs.swmansion.com/react-native-gesture-handler/, since the whole drawing relies on `react-native-gesture-handler`. Please follow the setup instructions on their docs and let me know if it still doesn't work afterwards.

I just tried creating a new react native project with this library and it is working (there were only some small syntax issues in the example, which I just fixed).

**Comment by Anubhavshakya** [comment]
First of all thank you for quick response, I have setup this project using expo managed. I have tried many ways but it still does not draw anything. I am attaching the UI and my Code. Can you provide me working example repo ?? so that it can help me.
![WhatsApp Image 2022-02-02 at 10 56 24 PM](https://user-images.githubusercontent.com/38704248/152206886-2241d8d2-3946-4c5e-bee2-9f41c4996fb2.jpeg)

`import React, { useRef } from 'react';
import { Draw, DrawRef } from "@benjeau/react-native-draw";

export default function App() {
  const drawRef = useRef<DrawRef>(null);

  const removeLastPath = () => {
    drawRef.current?.undo();
  }

  const clearDrawing = () => {
    drawRef.current?.clear();
  }

  // ... for more ref functions, look below

  return (
    <Draw
      ref={drawRef}
      height={600}
      initialValues={{
        color: "#B644D0",
        thickness: 10,
        opacity: 0.5,
        paths: []
      }}
    />
  )
}`

**Comment by BenJeau** [comment]
Have you tried running the [example](https://github.com/BenJeau/react-native-draw/tree/master/example)? (although the code inside may seem a bit weird, just picture that the objects in the `data` variable in the [data.ts](https://github.com/BenJeau/react-native-draw/blob/master/example/src/screens/data.ts) file as props to the component, more specifically the `props` object of the `data` variable)

It is using Expo and should be working on Android and iOS (web has a problem with one of the dependencies).

If that is not enough, I can create a small repo with only the example.

**Reply by mhumza37dev** [reply]
Nothing happens when i try to draw on canvas.

https://user-images.githubusercontent.com/68570316/180904283-6efe2314-5a48-476f-b91d-5289700614d2.mp4

**Comment by Anubhavshakya** [comment, marked as answer]
When I run the example then I got babel runtime error. I have try to solve it by googling but none of them working.
![Screenshot Capture - 2022-02-03 - 18-53-49](https://user-images.githubusercontent.com/38704248/152351624-fc871804-5c74-4732-978d-18b733bbe2d2.png)

**Reply by BenJeau** [reply]
Just making sure, have you run yarn install at the root of the repository?

**Reply by Anubhavshakya** [reply]
Oh god !! that was my mistake.... I have not run yarn install at root of repository. It is now working..... Thank you so much.
 But I have one question, so How can i use it in my existing project ??? can you Just give me One example where it uses in project ?? Is I need to put whole code in My project???

**Reply by BenJeau** [reply]
1. You need to install this npm package

```
yarn add @BenJeau/react-native-draw
```

2. You only need to install these dependencies too (if you don't already have them installed) [react-native-gesture-handler](https://github.com/software-mansion/react-native-gesture-handler), [@react-native-community/slider](https://github.com/callstack/react-native-slider) and [react-native-svg](https://github.com/react-native-svg/react-native-svg).

Since you are using expo, you should just run the following command:

```
expo install @react-native-community/slider react-native-gesture-handler react-native-svg
```

> You may need to follow these extra steps for `react-native-gesture-handler` mentioned here https://docs.expo.dev/versions/latest/sdk/gesture-handler/

3. Then, in your application you would only need to use the component, here a very simple usage of this component:

```tsx
import React from 'react';
import { Draw } from '@benjeau/react-native-draw';

export default () => <Draw />;
```

**Reply by Anubhavshakya** [reply]
I have checked by creating a new project, I have setup all required packages according to their installation guide. as a separate package '@benjeau/react-native-draw' is not working. I request you to try it as a package on existing or new project.

**Reply by BenJeau** [reply]
I have created a new project and it is working, have you maybe forgotten to import the following at the top of your file? (as per the extra steps for https://docs.expo.dev/versions/latest/sdk/gesture-handler/)

```tsx
import 'react-native-gesture-handler';
```

**Reply by Anubhavshakya** [reply]
Here is my Code
![Screenshot from 2022-02-03 22-09-39](https://user-images.githubusercontent.com/38704248/152387206-065f18d6-78e2-44a7-a3bd-05ff7ec8c32f.png)

**Reply by BenJeau** [reply]
I assume you are trying it on Android and that you cannot draw? This should help you https://stackoverflow.com/a/70950177/11006837 - you need to do this since you are using `Expo`

Here's a screenshot of it working:

<img width="1705" alt="Screen Shot 2022-02-03 at 11 53 10 AM" src="https://user-images.githubusercontent.com/22248828/152389568-039c69b5-5487-4e32-a75a-491b7ad68ad8.png">

**Reply by Anubhavshakya** [reply]
Finally Problem solved.... Thanks for your help and Thanks for this amazing package. I am working on a project in which I will use it.... Can I use this package in js rather than typescript ???

**Reply by BenJeau** [reply]
No problem! glad I could help, yes it should work in javascript too
